### DISCUSSION TITLE
Templating extension

### DISCUSSION THREAD

**Discussion by clyra** [espanso/espanso#1271] [Feature requests and Ideas]
**Is your feature request related to a problem? Please describe.**

Forms are cool, but it could be even better if we could use some sort of templating, instead of just inserting the values at the place holder markers. One user complained that he can't use the same value twice in forms (https://www.reddit.com/r/espanso/comments/tc5elr/insert_same_value_in_multiple_places_in_form/) and another about conditional expansions (https://www.reddit.com/r/espanso/comments/tti6eh/conditional_expansion/). I think both problems could be solved with a templating engine.

**Describe the solution you'd like**
A templating extension which should accept a template (file or inline) and render it using variables from other extensions. A quick googling showed "tera" (https://tera.netlify.app) which looks nice as it was inspired by jinja2 syntax.

**Describe alternatives you've considered**
The templating can be done outside Espanso with python+jinja or whatever, but that would require installing python, jinja or whatever templating system one wishes to use, but that would exclude a lot of users who just wishes to write pre-canned emails, reports, html snippets and so on...

**Additional context**

**Comment by federico-terzi** [comment]
Hey @clyra,

Thank you for the feedback! I see how this could be useful, though I'm wondering whether it would be better to extend the internal espanso syntax instead of supporting a completely different one. We'll investigate!

Cheers :)

**Comment by NicolasWebDev** [comment]
Happy new year!

Any news on that front? Is it something maybe on the roadmap @federico-terzi ?
